shell/unix/load.bash

```diff
@@ -0,0 +1,89 @@
+# Alternative source to load, uses specific bash extensions to the POSIX shell (BASH_SOURCE, <<<) hence the .bash extension
+# Usage in ~/.bashrc: source ~/path/to/volta/load.bash
+# Example: source ~/.volta/load.bash
+# Or: [[ -s ~/.volta/load.bash ]] && source ~/.volta/load.bash
+
+# TODO: test on windows, window WSL and osx. hahahaha.
+
+
+# Example usage in bash:
+# $ source .volta/load.bash # source it directly
+
+# $ echo $VOLTA_HOME # was VOLTA_HOME defined?
+# /home/mcarifio/.volta
+
+# $ tr ':' '\n' <<< $PATH|grep volta # was the volta bin added to PATH?
+# /home/mcarifio/.volta/bin
+
+# $ volta --version # does the function run?
+# 0.6.3
+# volta which node # does $VOLTA_HOME/shim ... shim?
+# /home/mcarifio/.volta/tools/image/node/12.10.0/6.10.3/bin/node
+# $ unset VOLTA_HOME  # VOLTA_HOME _must_ be set
+# $ volta --version
+# -bash: VOLTA_HOME: Environment variable VOLTA_HOME undefined. Please export it.
+# $ export VOLTA_HOME=$HOME/.volta # get it back
+# $ node --version
+# v12.10.0
+ 
+
+
+# Don't pollute current environment with local functions _*
+trap "unset -f _volta_setup _volta_error" EXIT
+
+_volta_error() {
+    echo $* 1>&2
+    return 1  # not exit otherwise you exit the current session
+}
+
+# Expecting bash.
+[[ -z "${BASH}" ]] && _volta_error "Expecting bash."
+
+# Make _setup a function so you don't pollute the environment with working variables like `me` and `here`
+#  and you don't have to keep track of them.
+_volta_setup() {
+    local me=$(realpath ${BASH_SOURCE})
+    local here=${me%/*}
+    export VOLTA_HOME=${VOLTA_HOME:-${here}}
+    if [[ -d ${VOLTA_HOME} ]] ; then
+        local bin=${VOLTA_HOME}/bin
+        [[ -d ${bin} ]] || _volta_error "Expecting ${bin}, not found."
+        # Add to PATH iff not already there. Next line is hack city.
+        if ! tr ':' '\n' <<< ${PATH} | grep --silent --no-messages ${bin} ; then
```

Comment:
This line is pretty impossible to understand. If we're trying to check that the `bin` isn't already on the path, can we do that with a regex match, or even a simpler pattern match:

```
if [[ "${PATH}" != *$bin* ]]; then
```

Or something like that? I'll admit to being a novice at shell scripting, so if that's not possible, please let me know :smile:

Reply:
Yes. You could do this with a re match of ':\b*${bin}:\b*' on PATH. Bash regular expressions are also a little persnicky, so a reader will need to google stuff regardless.

Reply:
tr gone, re instead